src/app/app.ts

```diff
@@ -123,6 +123,16 @@ class AppImpl implements App {
             actionType = 'create';
         }
 
+        const adminClient = newMMClient(this.mmOptions).asAdmin();
+
+        // add bot to team and channel
+        const botUserID = this.context.bot_user_id;
+        const addToTeamReq = adminClient.addToTeam(this.context.team_id, botUserID);
+        await tryPromiseWithMessage(addToTeamReq, 'Failed to add bot to team');
+
+        const addToChannelReq = adminClient.addToChannel(botUserID, this.context.channel_id);
+        await tryPromiseWithMessage(addToChannelReq, 'Failed to add bot to channel');
```

Comment:
This seems like a hack to me. An app should be contained by it's own channel membership and not add itself to all channels, even though it can do that. Requiring `post_all_as_bot` is more honest.

Reply:
> Arguably, if the acting user doesn't have permission to add users to the current channel, the shouldn't have access to create subscriptions in the current channel. 

That the important point to me.

Reply:
@hanzei The solution I mentioned uses the acting user's token to do the add the bot to the channel. So if the acting user does not have permissions to add the bot to the channel, then the subscription add would/should fail. No non-permissive channel adding going on here, as long as we use the acting user token to do the add.

Reply:
I have changed the code to:

1. check that the acting user is a MM system admin
2. the MM client now uses the acting_user_id, not the general admin access_token